examples/packetdump.rs

```diff
@@ -44,25 +45,51 @@ fn handle_tcp_packet(interface_name: &str, source: IpAddr, destination: IpAddr,
     // place in both TCP and UDP headers, we cheat here
     let udp = UdpPacket::new(packet);
     if let Some(udp) = udp {
-        println!("[{}]: TCP Packet: {}:{} > {}:{}; length: {}", interface_name, source,
-                    udp.get_source(), destination, udp.get_destination(), packet.len());
+        println!("[{}]: TCP Packet: {}:{} > {}:{}; length: {}", 
+                 interface_name, 
+                 source,
+                 udp.get_source(), 
+                 destination, 
+                 udp.get_destination(), 
+                 packet.len());
     } else {
         println!("[{}]: Malformed TCP Packet", interface_name);
     }
 }
 
+fn handle_icmp_packet(interface_name: &str, source: IpAddr, destination: IpAddr, packet: &[u8]) {
+    // Since we only look at source and destination ports, and these are located in the same
+    // place in both TCP and UDP headers, we cheat here
+    let icmp = IcmpPacket::new(packet);
```

Comment:
The `IcmpPacket::new()` method will return an `Option<IcmpPacket>` - you'll notice that in the other `handle_*_packet` methods it uses an `if let` statement to get the `IcmpPacket` out. I believe this is the cause of your errors.